### PR TITLE
Added DelegatingMetricData & TestMetricData Classes

### PR DESCRIPTION
**Pull request by ThomasJazz**
Hello! This is my first attempt to contribute to an open source project, so pls be nice :). I have read through the contributor steps, but I apologize in advance if I have missed anything.

- Changes should address issue: https://github.com/open-telemetry/opentelemetry-java/issues/6195
- I used DelegatingSpanData as a guide for this. I am not familiar enough with the OTEL codebase yet to know the full big picture, so if there's any code in there that doesn't make sense, that's why

### CODE REVIEW

File: sdk/testing/src/main/java/io/opentelemetry/sdk/testing/metrics/TestMetricData.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+public abstract class TestMetricData implements MetricData {
+  public static Builder builder() {
+    return new AutoValue_TestMetricData.Builder()
+        .setResource(Resource.empty())
+        .setName("name")
+        .setInstrumentationScopeInfo(InstrumentationScopeInfo.empty())
+        .setDescription("description")
+        .setUnit("1")
+        .setData(ImmutableSummaryData.empty())
+        .setType(MetricDataType.SUMMARY);
```
Comment (jack-berg) [inline sdk/testing/src/main/java/io/opentelemetry/sdk/testing/metrics/TestMetricData.java:32]: Not sure setting data and type to arbitrary selections is better than omitting them.

Also, the value of this is limited without usable `Data` and `PointData` implementations.

@jkwatson WDYT about adding static create methods to all the interfaces in the [io.opentelemetry.sdk.metrics.data](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data)? These could return the [internal immutable implementations](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/data), without requiring that we make all those classes public.

**Review by jack-berg — COMMENTED** [review]
Thanks for working on this @ThomasJazz!

**Comment by jack-berg** [comment]
@ThomasJazz I'm going to close this because there hasn't been any activity in several months. Feel free to re-open if you pick this up again.

File: sdk/testing/src/main/java/io/opentelemetry/sdk/testing/metrics/TestMetricData.java (hunk 1 above)
Comment (chukunx) [reply sdk/testing/src/main/java/io/opentelemetry/sdk/testing/metrics/TestMetricData.java:32]: hey @jack-berg it looks like all the immutable implementations already have public `create` methods. do you mind share a bit more detail on the last comment above? 

would it look better to just return the `AutoValue_TestMetricData.Builder()` form the `builder()` method?

**Pull request closed without merging**